src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -810,53 +899,66 @@ CopyLocalDataIntoShards(Oid distributedRelationId)
 	econtext->ecxt_scantuple = slot;
 
 	copyDest =
-		(DestReceiver *) CreateCitusCopyDestReceiver(distributedRelationId,
+		(DestReceiver *) CreateCitusCopyDestReceiver(destinationDistributedRelationId,
 													 columnNameList, estate,
 													 stopOnFailure);
 
 	/* initialise state for writing to shards, we'll open connections on demand */
 	copyDest->rStartup(copyDest, 0, tupleDescriptor);
 
-	/* begin reading from local table */
-	scan = heap_beginscan(distributedRelation, GetActiveSnapshot(), 0, NULL);
-
-	oldContext = MemoryContextSwitchTo(GetPerTupleMemoryContext(estate));
-
-	while ((tuple = heap_getnext(scan, ForwardScanDirection)) != NULL)
+	/* iterate on the local relations, copy each into the destination relation */
+	foreach(sourceRelationCell, sourceRelationList)
```

Comment:
> Burak's PR should also address this.

@marcocitus Well, yes without that PR, I still get the error. Could you please help me understand what is happening?

[Here](https://github.com/citusdata/citus/compare/partition_distributed_table_create_table...connection_error?expand=1) is the roughly the changes that I'm planning to make on top of this branch (i.e., change `CopyLocalDataIntoShards ` to be as it is and do not alternate between copy & DDL). When you swith to that branch, you'll see the following:

```SQL

CREATE TABLE partitioning_test(id int, time date) PARTITION BY RANGE (time);
 
-- create its partitions
CREATE TABLE partitioning_test_2009 PARTITION OF partitioning_test FOR VALUES FROM ('2009-01-01') TO ('2010-01-01');
CREATE TABLE partitioning_test_2010 PARTITION OF partitioning_test FOR VALUES FROM ('2010-01-01') TO ('2011-01-01');

-- load some data and distribute tables
INSERT INTO partitioning_test VALUES (1, '2009-06-06');
INSERT INTO partitioning_test VALUES (2, '2010-07-07');
INSERT INTO partitioning_test VALUES (3, '2009-06-07');
INSERT INTO partitioning_test VALUES (4, '2010-07-08');
INSERT INTO partitioning_test_2009 VALUES (5, '2009-09-09');
INSERT INTO partitioning_test_2010 VALUES (6, '2010-03-03');
INSERT INTO partitioning_test_2009 VALUES (7, '2009-09-09');
INSERT INTO partitioning_test_2010 VALUES (8, '2010-03-03');

SET citus.shard_count TO 4;
SET citus.shard_replication_factor TO 1;

-- this should error out given that parent of the partition is not distributed
SELECT create_distributed_table('partitioning_test_2010', 'id');

-- this should suceed
SELECT create_distributed_table('partitioning_test', 'id');

ERROR:  cannot establish new placement connection when DML has been executed on existing placement connection for a colocated placement
```

However, when I try similar stuff (without partitioning, using transaction blocks explicitly), the similar things works:
```SQL
CREATE TABLE test_0 (a int, b int);
INSERT INTO test_0 SELECT s, s FROM generate_series(0, 100000) s;
SET citus.shard_count TO 4;
SELECT create_distributed_table('test_0', 'a');

CREATE TABLE test_1 (a int, b int);
CREATE TABLE test_2 (a int, b int);

INSERT INTO test_1 SELECT s, s FROM generate_series(0, 100000) s;
COPY test_1 TO '/tmp/data.csv' WITH CSV;

BEGIN;

SET citus.shard_count TO 4;
SELECT create_distributed_table('test_1', 'a');
SELECT create_distributed_table('test_2', 'a');

CREATE INDEX i1 ON test_1 (a);
CREATE INDEX i2 ON test_2 (a);

COPY test_1 FROM '/tmp/data.csv' WITH CSV;


COPY test_2 FROM '/tmp/data.csv' WITH CSV;

```

So, do you have any ideas on `ERROR:  cannot establish new placement connection when DML has been executed on existing placement connection for a colocated placement` ?

Shall I wait for Burak's PR for that?